editor/src/components/assets.ts

```diff
@@ -40,6 +41,50 @@ export function getAllProjectAssetFiles(
   return allProjectAssets
 }
 
+function getSHA1Checksum(contents: string): string {
+  return new sha1().update(contents).digest('hex')
+}
+
+export function getProjectContentsChecksums(tree: ProjectContentTreeRoot): GithubChecksums {
+  const contents = treeToContents(tree)
+
+  const checksums: GithubChecksums = {}
+  Object.keys(contents).forEach((filename) => {
+    const file = contents[filename]
+    if (isTextFile(file)) {
+      checksums[filename] = getSHA1Checksum(file.fileContents.code)
+    } else if (isAssetFile(file) && file.base64 != undefined) {
+      checksums[filename] = getSHA1Checksum(file.base64)
+    }
+  })
+
+  return checksums
+}
+
+export function deriveGithubFileChanges(
+  projectChecksums: GithubChecksums,
+  githubChecksums: GithubChecksums | null,
+): GithubFileChanges | null {
+  if (githubChecksums == null) {
+    return null
+  }
+
+  const projectFiles = Object.keys(projectChecksums)
+  const githubFiles = Object.keys(githubChecksums)
```

Comment:
could be turned into a Set to speed up the includes calls?

Reply:
https://github.com/concrete-utopia/utopia/pull/2733/commits/6f95b442e8a942bdc7aff83b506c23a09936bbe6